src/Widgets/HeaderBar.vala

```diff
@@ -224,47 +224,58 @@ public class Camera.Widgets.HeaderBar : Gtk.HeaderBar {
     }
 
     public void add_camera_option (Gst.Device camera) {
-        var menuitem = new Gtk.RadioMenuItem.with_label (null, camera.display_name);
-        menuitem.set_data<Gst.Device> ("camera", camera);
-        camera_options.append (menuitem);
-
-        int i = (int) camera_options.get_children ().length () - 1;
-        if (i > 0) {
-            var el = camera_options.get_children ().nth_data (0) as Gtk.RadioMenuItem;
-            menuitem.join_group (el);
-        }
-        menuitem.active = true;
-        menuitem.activate.connect (() => {
-            if (menuitem.active) {
-                request_camera_change (menuitem.get_data<Gst.Device> ("camera"));
-            }
-        });
-        menuitem.show ();
+        // var menuitem = new MenuItem (camera.display_name, null);
+        // menuitem.set_data<Gst.Device> ("camera", camera);
+        // camera_options.append_item (menuitem);
+
+        // int i = (int) camera_options.get_n_items () - 1;
+        // if (i > 0) {
+        //     var el = camera_options.get_children ().nth_data (0) as MenuItem;
+        //     menuitem.join_group (el);
+        // }
+        // menuitem.active = true;
+        // menuitem.activate.connect (() => {
+        //     if (menuitem.active) {
+        //         request_camera_change (menuitem.get_data<Gst.Device> ("camera"));
+        //     }
+        // });
+        // menuitem.show ();
+        camera_options.append (camera.display_name, "win.change-camera('%s')".printf (camera.name));
```

Comment:
```suggestion
        camera_options.append (camera.display_name, "win.change_camera('%s')".printf (camera.name));
```

Alternatively, you can use `GLib.Action.print_detailed_name()`, using the constants defined on the MainWindow:

```suggestion
        camera_options.append (camera.display_name, GLib.Action.print_detailed_name (
            MainWindow.ACTION_PREFIX + MainWindow.ACTION_CHANGE_CAMERA,
            new GLib.Variant.string (camera.name)
        ));
```